lightly/utils/embeddings_2d.py

```diff
@@ -53,13 +59,22 @@ def transform(self, X: np.ndarray):
         Returns:
             Numpy array of n x p datapoints where p <= d.
 
+        Raises:
+            ValueError: If PCA was not fitted before.
         """
+        if self.mean is None or self.w is None:
+            raise ValueError("PCA not fitted yet. Call fit() before transform().")
         X = X.astype(np.float32)
         X = X - self.mean + self.eps
-        return X.dot(self.w)[:, : self.n_components]
+        transformed = X.dot(self.w)[:, : self.n_components]
+        return np.asarray(transformed, dtype=np.float32)
```

Comment:
Is this needed because numpy could widen it to `np.float64`? Or why can we not do `transformed: NDArray[np.float32] = X.dot(...)`